tests/callbacks/test_tqdm_progress_bar.py

```diff
@@ -276,20 +263,21 @@ def test_tqdm_progress_bar_default_value(tmpdir):
     trainer = Trainer(default_root_dir=tmpdir)
     assert trainer.progress_bar_callback.refresh_rate == 1
 
-    trainer = Trainer(default_root_dir=tmpdir, progress_bar_refresh_rate=None)
-    assert trainer.progress_bar_callback.refresh_rate == 1
-
 
 @mock.patch.dict(os.environ, {"COLAB_GPU": "1"})
 def test_tqdm_progress_bar_value_on_colab(tmpdir):
     """Test that Trainer will override the default in Google COLAB."""
     trainer = Trainer(default_root_dir=tmpdir)
     assert trainer.progress_bar_callback.refresh_rate == 20
 
-    trainer = Trainer(default_root_dir=tmpdir, progress_bar_refresh_rate=None)
-    assert trainer.progress_bar_callback.refresh_rate == 20
+    trainer = Trainer(default_root_dir=tmpdir, callbacks=TQDMProgressBar())
+    assert trainer.progress_bar_callback.refresh_rate == 1  # FIXME: should be 20
```

Comment:
Is this a bug? Can be done in a separate PR.

Reply:
Yes and no :) As you provide manual the progress bar, the default value is set to 1. However, it would be better to set it to 20 in the COLAB env.

I believe a way to resolve this would be for this logic to live within the Callback directly as follows:

```py
import os
from typing import Optional

class TQDMProgressBar(ProgressBarBase):

    def __init__(self, refresh_rate: Optional[int] = None, process_position: int = 0):
        super().__init__()

        if refresh_rate is None:
            refresh_rate = 20 if os.getenv("COLAB_GPU", "1") == "1" else 1

        self._refresh_rate = refresh_rate
```

Reply:
Yes, exactly.

Reply:
Added it here #10542. And tested that `RichProgressBar` wouldn't require this setup

Reply:
Thanks @kaushikb11, let's get this merged so you can fix it on top